yjit/src/core.rs

```diff
@@ -487,6 +487,21 @@ impl IseqPayload {
         // Turn it into an iterator that owns the blocks and return
         version_map.into_iter().flatten()
     }
+
+    /// Return code address ranges used by the ISEQ
+    pub fn code_addrs(&self) -> impl Iterator<Item = (CodePtr, CodePtr)> {
+        let mut addrs: Vec<(CodePtr, CodePtr)> = vec![];
+        // TODO: support ISEQ entry
+        for version_list in self.version_map.iter() {
+            for block in version_list.iter() {
+                match (block.borrow().start_addr, block.borrow().end_addr) {
+                    (Some(start_addr), Some(end_addr)) => addrs.push((start_addr, end_addr)),
+                    _ => {}
+                }
+            }
+        }
+        return addrs.into_iter();
+    }
```

Comment:
This is maybe not that important but here you are allocating a vector and adding addresses to it. We might be able to get better performance if there is a way to produce an iterator directly without appending to a vector?

At the same time I understand that this is a first prototype and maybe how fast the code GC runs is not a priority at this point. Could add a TODO or FIXME comment to mention this.